core/debugger/remote_debugger_peer.cpp

```diff
@@ -163,7 +167,11 @@ Error RemoteDebuggerPeerTCP::connect_to_host(const String &p_host, uint16_t p_po
 	const int tries = 6;
 	const int waits[tries] = { 1, 10, 100, 1000, 1000, 1000 };
 
-	tcp_client->connect_to_host(ip, port);
+	Error err = tcp_client->connect_to_host(ip, port);
+	if (err != OK) {
+		ERR_PRINT(vformat("Remote Debugger: Unable to connect to host '%s:%d'.", p_host, port));
+		return err;
+	}
```

Comment:
Check the error and return, as it was causing mobile to crash if the remote host was not set correctly.